src/python/library/tritonclient/grpc/_infer_input.py

```diff
@@ -86,6 +87,36 @@ def shape(self):
         """
         return self._input.shape
 
+    def validate_data(self):
+        """Validate input has data and input shape matches input data.
+
+        Returns
+        -------
+        None
+        """
+        # Input must set only one of the following fields: '_raw_content',
+        # 'shared_memory_region' in '_input.parameters'
+        cnt = 0
+        cnt += self._raw_content != None
+        cnt += "shared_memory_region" in self._input.parameters
+        if cnt != 1:
+            return
```

Comment:
Shouldn't we return an error when more that one fields are specified in the inputs?

Reply:
This is handled by another existing function after ValidateData(). I didn't want to duplicate the check in ValidateData().